src/EPR.SubsidiaryBulkUpload.Application.UnitTests/Services/NotificationServiceTests.cs

```diff
@@ -100,4 +101,148 @@ public async Task SetErrorStatus_ShouldSerializeErrorsAndSetStringInRedisAndLogI
+                }
+        };
+
+        var json = JsonSerializer.Serialize(errors);
+        _redisDatabaseMock.Setup(db => db.StringGetAsync(It.IsAny<RedisKey>(), It.IsAny<CommandFlags>()))
+            .ReturnsAsync((RedisValue)json);
+
+        // Act
+        var result = await _notificationService.GetNotificationErrorsAsync(key);
+
+        // Assert
+        Assert.IsNotNull(result);
+        Assert.AreEqual(2, result.Errors.Count);
+        Assert.AreEqual(1, result.Errors[0].FileLineNumber);
+        Assert.AreEqual("Content1", result.Errors[0].FileContent);
+        Assert.AreEqual("Message1", result.Errors[0].Message);
+        Assert.AreEqual(6, result.Errors[0].ErrorNumber);
+        Assert.IsTrue(result.Errors[0].IsError);
+        Assert.AreEqual(2, result.Errors[1].FileLineNumber);
+        Assert.AreEqual("Content2", result.Errors[1].FileContent);
+        Assert.AreEqual("Message2", result.Errors[1].Message);
+        Assert.IsFalse(result.Errors[1].IsError);
+        Assert.AreEqual(9, result.Errors[1].ErrorNumber);
+
+        _loggerMock.Verify(
+            x => x.Log(
+                It.Is<LogLevel>(l => l == LogLevel.Information),
+                It.IsAny<EventId>(),
+                It.Is<It.IsAnyType>((v, t) => v.ToString() == $"Redis errors response key: {key} errors: {json}"),
+                It.IsAny<Exception>(),
+                It.Is<Func<It.IsAnyType, Exception, string>>((v, t) => true)),
+            Times.Once);
+    }
+
+    [TestMethod]
+    public async Task ClearRedisKeyAsync_KeyExists_ShouldLogKeyDeleted()
+    {
+        // Arrange
+        var key = "testKey";
+        _redisDatabaseMock
+            .Setup(db => db.KeyDeleteAsync(key, It.IsAny<CommandFlags>()))
+            .ReturnsAsync(true);
+
+        // Act
+        await _notificationService.ClearRedisKeyAsync(key);
+
+        // Assert
+        _redisDatabaseMock.Verify(db => db.KeyDeleteAsync(key, It.IsAny<CommandFlags>()), Times.Once);
+        _loggerMock.Verify(
+            log => log.Log(
+                LogLevel.Information,
+                It.IsAny<EventId>(),
+                It.Is<It.IsAnyType>((v, t) => v.ToString().Contains("Redis key testKey deleted successfully.")),
+                null,
+                It.IsAny<Func<It.IsAnyType, Exception, string>>()),
+            Times.Once);
+    }
+
+    [TestMethod]
+    public async Task ClearRedisKeyAsync_KeyDoesNotExist_ShouldLogKeyNotFound()
+    {
+        // Arrange
+        var key = "testKey";
+        _redisDatabaseMock
+            .Setup(db => db.KeyDeleteAsync(key, It.IsAny<CommandFlags>()))
+            .ReturnsAsync(false);
+
+        // Act
+        await _notificationService.ClearRedisKeyAsync(key);
+
+        // Assert
+        _redisDatabaseMock.Verify(db => db.KeyDeleteAsync(key, It.IsAny<CommandFlags>()), Times.Once);
+        _loggerMock.Verify(
```

Comment:
This can be simpolified to
`_loggerMock.VerifyLog(x => x.LogWarning("Redis key testKey not found."), Times.Once);`

Reply:
changed